docs: add docs for vulnerability related commands (#3362)
canonical/ubuntu-pro-client: docs/explanations/how_to_interpret_output_of_pro_vulnerability_show.rst

   vim          fixed   2:7.4.1689-3ubuntu1.5  2:7.4.1689-3ubuntu1.5+esm19  esm-infra
   vim-common   fixed   2:7.4.1689-3ubuntu1.5  2:7.4.1689-3ubuntu1.5+esm19  esm-infra
   vim-runtime  fixed   2:7.4.1689-3ubuntu1.5  2:7.4.1689-3ubuntu1.5+esm19  esm-infra
   vim-tiny     fixed   2:7.4.1689-3ubuntu1.5  2:7.4.1689-3ubuntu1.5+esm19  esm-infra
   
   [RELATED USNs]
   VULNERABILITY  TITLE                AFFECTED INSTALLED PACKAGES
   USN-6270-1     Vim vulnerabilities  vim

Let's break it down the output of this command. We start by telling you some basic information for
the CVE:

* **Public URL**: The ubuntu security page for this CVE
* **Published at**: The published date for the CVE
* **Ubunut vulnerability data published at**: The published for the vulnerability data we consume to
  generate this output
* **APT package information updated at**: The last time the APT state was updated in the system
  (i.e. running an apt install operation)
* **Ubuntu priority**: The ubuntu priority for this CVE
* **CVSS score**: The CVSS score of the CVE
* **CVESS severity**: The CVSS severity of the CVE

After that, we display the CVE description and related notes (if the CVE has notes).

> The next block is now displaying which installed packages in the machine are affected by the CVE.
> Whe are using table to display those affected packages where:

Reviewer: ```suggestion
The next block shows which packages installed on the machine are affected by the CVE.
We use a table to display the affected packages with the following columns:
```

* **NAME**: The package name
* **STATUS**: The CVE status for this package
* **CURRENT VERSION**: The current installed version of the package

> * **FIXED BY**: The version of that package that fix the CVE issue

Reviewer: ```suggestion
* **FIXED BY**: The version of the package that fixes the CVE
```

> * **FIX AVAILABLE FROM**: The pocket where the fix version can be found on

Reviewer: ```suggestion
* **FIX AVAILABLE FROM**: The pocket where the fix version can be found
```

> Finally, we also display the related USNs to the CVE, in case you want to take a look on them as
> well.

Reviewer: ```suggestion
Finally, we also display any USNs related to the CVE.
```

Reviewer: Should we link here to the page about related USNs?

Author: No, this is a different concept. Our related USNs documentation tell about a relation between USNs and other USNs, no CVEs to USNs

Reviewer: Yeah you're right, I was having a moment lol

What if the CVE doesn't affect my system ?
==========================================

> If the CVE doesn't affect your system, running the ``pro vulnerability show`` will show

Reviewer: ```suggestion
If the CVE doesn't affect your system, running ``pro vulnerability show`` will show
```

> you the an output like this:

Reviewer: ```suggestion
you an output like this:
```

.. code-block:: text

    ● CVE-2024-7542
    Public URL: https://ubuntu.com/security/CVE-2024-7542
    Ubuntu vulnerability data published at: 2024-10-22 23:37:41
    APT package information updated at: 2024-10-23 19:37:11
    
    CVE-2024-7542 does not affect your system.

Note that we directly inform you that the CVE doesn't affect your system.

What if some packages cannot be fixed for a CVE?
================================================

When that happens, our affected installed packages table will inform you about that.
For example, here is the table output for the situation where some packages cannot be
fixed:

.. code-block:: text

    NAME                  STATUS      CURRENT VERSION          FIXED BY  FIX AVAILABLE FROM
    libpython3.5          vulnerable  3.5.2-2ubuntu0~16.04.13  -         no-fix
    libpython3.5-minimal  vulnerable  3.5.2-2ubuntu0~16.04.13  -         no-fix
    libpython3.5-stdlib   vulnerable  3.5.2-2ubuntu0~16.04.13  -         no-fix
    python3.5             vulnerable  3.5.2-2ubuntu0~16.04.13  -         no-fix
    python3.5-minimal     vulnerable  3.5.2-2ubuntu0~16.04.13  -         no-fix

> Note that the package status show that it is **vulnerabable** and that the **FIXED BY** and

Reviewer: ```suggestion
The status column shows the package is **vulnerabable**, but the **FIXED BY** and
```

> **FIX AVAILABLE FROM** column will tell you that no fix is available for the package.

Reviewer: ```suggestion
**FIX AVAILABLE FROM** column tell you that no fix is available for the package.
```

What if I want to see a USN instead ?
=====================================

> To see information about a USN, you just need to provide a USN name instead. For example,

Reviewer: ```suggestion
To see information about a USN, you can provide a USN name instead. For example,
```

    pro vulnerability show USN-4976-2

Assuming you are affected by the USN, you might see an output like this:

.. code-block:: text

    Public URL: https://ubuntu.com/security/notices/USN-4976-2
    Published at: 2022-09-07 15:22:39
    Ubuntu vulnerability data published at: 2024-10-22 23:37:41
    APT package information updated at: 2024-10-23 19:37:11
    
    [DESCRIPTION]
    USN-4976-1 fixed a vulnerability in Dnsmasq. This update provides
    the corresponding update for Ubuntu 16.04 ESM.
    
    Dnsmasq has been updated to 2.79-1 for Ubuntu 16.04 ESM in order to fix
    some security issues.
    
    Original advisory details:
    
     Petr Mensik discovered that Dnsmasq incorrectly randomized source ports in
     certain configurations. A remote attacker could possibly use this issue to
     facilitate DNS cache poisoning attacks.
    
    
    [AFFECTED INSTALLED PACKAGES]
    NAME          CURRENT VERSION         FIXED BY                    FIX AVAILABLE FROM
    dnsmasq-base  2.75-1ubuntu0.16.04.10  2.79-1ubuntu0.16.04.1+esm1  esm-infra
    
    [RELATED CVEs]
    VULNERABILITY  PRIORITY  AFFECTED INSTALLED PACKAGES
    CVE-2021-3448  low       dnsmasq

> Note that the information presented here is similar to the one display for the CVE.

Reviewer: ```suggestion
The information presented here is similar to the output displayed for CVEs.
```

> However, we are not displaying information that is only tied to a CVE, such as

Reviewer: ```suggestion
However, we do not display information that is only relevant to a CVE, such as
```
